extensions/amp-base-carousel/0.1/carousel.js

```diff
@@ -287,6 +287,17 @@ export class Carousel {
     /** @private {!Axis} */
     this.axis_ = Axis.X;
 
+    /**
+     * Whether slides are laid out in the forwards or reverse direction. When
+     * using rtl (right to left), this should be false. This is used to set the
+     * transforms for slides and spacers correctly when the flex direction is
+     * reversed due to a rtl direction. TODO(sparhami) is there some way we
+     * could get this to work without needing to be explicitly told what the
+     * direction is?
+     * @private {boolean}
+     */
+    this.forwards_ = true;
```

Comment:
Does your code handle the case where the full document has `dir="rtl"` i.e. `<html dir="rtl">`? `dom.js#isRTL` should be able to handle that detection for this case. Or is that handled by the browser's flex box logic like you mentioned earlier?

https://github.com/ampproject/amphtml/blob/98c0db6fa6d2a25c004b026908f3a040d1dcf6ff/src/dom.js#L675-L685

Reply:
Yep, it is handled by the browser's flex logic (which flips based on the rtl-ness of the given subtree).